client/src/components/_common/Button/Button.jsx

```diff
@@ -39,6 +39,33 @@ const Button = ({
     }
   }
 
+  // Manage prop warnings
+  /* eslint-disable no-console */
+  if (type === 'link' && size) {
+    size = '';
+    // Component will work, except `size` is ineffectual
+    console.warn('A <Button type="link"> ignores `size` prop.');
+  }
+  if (type === 'primary' && size === 'small') {
+    type = 'secondary';
+    // Component will work, except `type` is ineffectual
+    console.error(
+      'A <Button type="primary" size="small"> is not allowed. ' +
+        'Using `type="secondary"` instead.'
+    );
+  }
+  if (type !== 'link' && !size) {
+    size = 'short';
+    // Component will work, except `size` is auto-set
+    console.debug(
+      'A <Button> that is not `type="link"` and has no `size` ' +
+        'is automatically assigned `size="short"`.'
+    );
+  }
+  /* eslint-enable no-console */
+
+  const buttonRootClass = styles['root'];
```

Comment:
**Intent:** Rename `c-button` to `root` so CSS module class names are simple (and consistent).

<details>Unique names are only necessary for global styles, which Portal does not often need (because it has CSS modules). Core-CMS uses all global styles, so unique names are used there. Core-Styles serves Core-CMS, and will serve Core-Portal and more, so it also uses unique names.</details>